src/spec/integration/import.spec.coffee

```diff
@@ -236,6 +320,71 @@ describe 'Import integration test', ->
         expect(p.name).toEqual en: "#{@newProductName}_changed"
         expect(p.slug).toEqual en: @newProductSlug
         expect(p.key).toEqual 'productKey'
+        expect(p.state).toBeUndefined
+        expect(p.masterVariant.key).toEqual 'variantKey'
+
+        done()
+      .catch (err) -> done _.prettify(err)
+
+    it 'should set default state on update when configured to do so, and product lacks state', (done) ->
+      csv =
+        """
+        productType,name,variantId,slug
+        #{@productType.id},#{@newProductName},1,#{@newProductSlug}
+        """
+      @importer.import(csv)
+      .then (result) =>
+        expect(_.size result).toBe 1
+        expect(result[0]).toBe '[row 2] New product created.'
+        csv =
+          """
+          productType,name,variantId,slug,key,variantKey
+          #{@productType.id},#{@newProductName+'_changed'},1,#{@newProductSlug},productKey,variantKey
+          """
+        im = new Import {
+          authConfig: authConfig
+          httpConfig: httpConfig
+          userAgentConfig: userAgentConfig
+          defaultState: 'previous-state'
+        }
+        im.matchBy = 'slug'
+        im.allowRemovalOfVariants = true
+        im.suppressMissingHeaderWarning = true
+        @client = im.client
+
+        im.import(csv)
+      .then (result) =>
+        expect(_.size result).toBe 1
+        expect(result[0]).toBe '[row 2] Product updated.'
+        service = TestHelpers.createService(project_key, 'productProjections')
+        request = {
+          uri: service
+          .where("productType(id=\"#{@productType.id}\")")
+          .staged true
+          .build()
+          method: 'GET'
+        }
+        @client.execute request
+      .then (result) =>
+        p = result.body.results[0]
+        expect(p.state).toBeUndefined
```

Comment:
Why is the state here `undefined`? You updated the product with a configured default state so it should create it right? And later you fetch the same product only with `.expand 'state'` added and somehow the state exists there.

Reply:
It looks like there's one too many imports going on there.  The product should be created with the default `import`, without state, then reimported with a new `import` and acquire state.  I need to investigate why it's behaving as you say here.

Reply:
Answer: The import was actually failing, but the done() in the catch clause wasn't causing the test to fail.  I wonder how many other tests are actually not reporting their failures due to this. :-(

Reply:
Was it because `_.prettify(err)` in `done()` returned a `false` value? Or is it a problem of Jasmine? 😢